### DISCUSSION TITLE
Adding a Cistercian Rite

### DISCUSSION THREAD

**Discussion by Augustinus-Altovadensis** [DivinumOfficium/divinum-officium#2522] [Ideas]
Hi, I'd like to add the traditional Cistercian rite. It would be basically the clone of the Monastic rite with some updates:
- Hymns before Urban VIII.
- Different antiphons etc.
- Minor differences (Deus, in adjutórium in the beginning of vigils, missing Confíteor in Compline etc.)
- Varying doxologies in minor Horas in certain periods
- etc.

I can do the work, but I would appreciate some pointers. To change the texts is not an issue, but to introduce a new rite... I've forked the repository and start playing around. A lot has worked right away (adding a Czech language etc.), but currently, I've hit sort of an impasse. 

I have changed the horas.pl CGI script, on line 1236, I've added a new line:
`  elsif ($version =~ /Ordo Cisterciense/i) { $suffix .= "Cist"; } `

After switching on the Cistercian version, the page stops responding and perl starts cycling somewhere, using 100% of the processor core, slowing down the system to the halt... I have copied all the files in the _Ordinarium_ folder with the "Cist" prefix for both languages I'm using, but that didn't help. 

Could someone point me in the right direction, what else do I need to change for the new rite to work? 

Thanks a lot!
O.A.M.D.G.

**Comment by mbab** [comment]
The problem for Competorium is with condition in specials.pl line 234.

BTW. Monastic *use* pre Urban hymns.

**Comment by Augustinus-Altovadensis** [comment]
> The problem for Competorium is with condition in specials.pl line 234.
> 
> BTW. Monastic _use_ pre Urban hymns.

Thanks for the reply!

So when I add the Cistercian version into the condition like this, it should be OK and the counter will stop increasing ad coelestia Regna? 
`  if ($version =~ /Monastic|Cistercian/i) { ...`

Yes, I know, the Hymns should be the same, but for some reason, they are not identical to what I have in my Cistercian breviary. Take Veni Creátor from the Terce of Pentecost + the Octave. The Monastic version in DivinumOfficium.com shows this for the second stanza:

`Qui díceris Paráclitus,
Altíssimi donum Dei,`

However the pre-Urban VIII. says this:

`Qui Paráclitus díceris,
donum Dei Altíssimi,`

Et cetera. Should I create a pull request for this, or is it OK for the official Monastic version? I don't have the resources, so I cannot check this myself in pre-1963 monastic hymnal...

**Comment by mbab** [comment]
> So when I add the Cistercian version into the condition like this, it should be OK and the counter will stop increasing ad coelestia Regna?
> ` if ($version =~ /Monastic|Cistercian/i) { ...`

For now yes, but I will try to fix this in other way.

> 
> Yes, I know, the Hymns should be the same, but for some reason, they are not identical to what I have in my Cistercian breviary. Take Veni Creátor from the Terce of Pentecost + the Octave. The Monastic version in DivinumOfficium.com shows this for the second stanza:
> 
> `Qui díceris Paráclitus, Altíssimi donum Dei,`
> 
> However the pre-Urban VIII. says this:
> 
> `Qui Paráclitus díceris, donum Dei Altíssimi,`

Our Monastic version is wrong. (We have not old version at all right now, I'll fix this.)

> 
> Et cetera. Should I create a pull request for this, or is it OK for the official Monastic version? I don't have the resources, so I cannot check this myself in pre-1963 monastic hymnal...
[Breviarium Monasticum 1930](https://archive.org/details/breviarium-monasticum-pars-ii/Breviarium%20Monasticum%20-%20Pars%20I)

**Comment by fiapps** [comment]
It occurs to me that if you were to use "Cistercian Monastic" for the name of the version, all of the code that matches `/Monastic/` would already apply to it without any changes. So you would only have to modify the code that applies to the 1963 Monastic Breviary and not to the Cistercian liturgy. This could simplify the work and the resulting code.

**Reply by Augustinus-Altovadensis** [reply]
BTW, I've tried it again, placing the Cistercian Monastic version _before_ the Benedictine Monastic, and it works now much better, so I will use it. Thanks for the tip! (I don't really care for the English name instead of Latin one, but yes, it simplifies the work, as well as the code).

**Comment by Augustinus-Altovadensis** [comment]
> It occurs to me that if you were to use "Cistercian Monastic" for the name of the version, all of the code that matches `/Monastic/` would already apply to it without any changes. So you would only have to modify the code that applies to the 1963 Monastic Breviary and not to the Cistercian liturgy. This could simplify the work and the resulting code.


Yes, you are right. However I need to account for the peculiarities of the Cistercian rite and therefore I need proper files for each Hora (and Calendar, which will be a huge pain...). And adding this suffix seems not to work in the moment, when I name it Cistercian office. 

BTW, is there any forum or chat with developers, well you, guys, where I can have some questions? I don't really think that bug reporting is the right place, but the Discussion section is empty, so unused?

I need to account for other things like the doubling of Antiphons (Cistercian rite never doubles any antiphones), missing R.br. in minor horas, one antiphon for Lauds etc. And of course, I could try to find out on my own, but it seems to me that many of you guys know the answer ouf of the bat, and several lines of text here can save me hours of reading the code... Thanks!

**Comment by mbab** [comment]
> BTW, is there any forum or chat with developers, well you, guys, where I can have some questions? I don't really think that bug reporting is the right place, but the Discussion section is empty, so unused?

**Discussion** are not used because they were opened yesterday. :-)
Please use them.

**Reply by Augustinus-Altovadensis** [reply]
Sure, my pleasure 🙂

Currently I have several things I'd like to sort out:

**1.  Doubling of the Antiphons.** Cistercian rite doesn't double at all. Removing it (conditionally) in Benedictus and Magnificat was a piece of cake, so were the minor hours. However, for the major hours, it still eludes me. I have tried the same trick as before - in _specials.pl_, in the _sub antetpsalm_, I have tried to add the following condition before and later after the Alleluia-routine:
`if ($version =~ /Cistercian/i) { $ant1 =~ $ant[0]; } `
But for some reason, it does nothing, in both positions. The Monastic version doubles always, so there is probably something I'm missing...

**2. One Antiphon for the Lauds.** Cistercian rite does have only one Antiphon for the Lauds, quite in the manner of the Alleluia Antiphon, but for the whole year (well, except for the Holy Triduum, where it copies the Roman rite, it just doesn't double, but first things first). I haven't even begun to solve this, the Alleluia routine might be a help, but if you have any pointers, I would be glad. 

**3. Responsorium breve.** R. br. is sung only in major hours in the Cistercian rite, which is the same like  the Monastic version. But in the Cistercian version, there is no Glória Patri in the responsories for Lauds and Vespers. Is there any way to fix this without having to rewrite all the responsories in the TXT files? 

There will be more, but this is enough for now. I will be glad for any help and pointing in the right direction. Thank you!

**Reply by mbab** [reply]
ad. 1.
in `sub antepsalm_mm`
replace line
`  if ($line[0]) { push(@s, "Ant. $line[0]"); $lastantiphon = $line[0]; }`
with
``` 
 if ($line[0]) {
    my $ant1 = ($version =~ /Cistercian/) ? substr($line[0], 0, index($line[0], '*')) : $line[0];
    push(@s, "Ant. $ant1"); $lastantiphon = $line[0];
  }
```

**Reply by mbab** [reply]
ad. 2.
Is it one of the antiphon used in Monastic or something different?  how it changes: weekday and/or tempora?

**Reply by mbab** [reply]
ad. 3.
if file `specials.pl`
add following line after 361 `if ($version =~ /monastic/i) {`
```
        if ($version =~ /Cistercian/) { $capit =~ s/&Gloria.*?_/_/s; }
```

**Reply by Augustinus-Altovadensis** [reply]
Thanks a lot! 
Ad 1. I couldn't find the line (maybe I need to sync with master...), but I managed to solve it by using the correct operator. I did a lot of programming in c and BASH, so there's a lot for me to learn in perl.

> ad. 2.
> Is it one of the antiphon used in Monastic or something different? how it changes: weekday and/or tempora?

Yes, it usually is one of the Monastic/Roman Antiphons. It does change a lot. There is a proper Antiphon for every Feria as well as for the Feasts, where it's usually the first Antiphon of the first Vespers. 
I haven't started with the Calendar yet, but I guess there will have to be a folder TemporaCist, CommuneCist etc., that will just be the copy of TemporaM and CommuneM (etc.), with the appropriate changes to the Roman rite (not to over-reference....)

I would say about 95% of the Calendar (and texts, Antiphons etc.) is identical to Monastic. There are also logically no doubles or semidoubles etc., but the ranks of the feasts have different names: 
- Festum sermonis major and minor for doubles
- Festum duarum missarum (MM. maj. and MM. min) for semidoubles, then 
- Festum xij. Lectionum
- Festum iij. Lect. 
- Commemoratio et Missa
- Commemoratio for the lowest rank. 
But I see that the variants are already there, so this should be possible to implement, shouldn't it? 

Ad 3. Works like charm, thanks!

**Reply by mbab** [reply]
ad. 2.
Introducing section like [Ant LaudesC] in monastic data files seems valid with default value to first line of [Ant Vespera].

If it is 95% introducing new directories looks unnecessary.
Try use existing existing Monastic data:
- use magic comments like "(rubrica Cistercian)" or "(rubrica Cistercian omittitur)"
- or create files like ex. SanctiM/08-20c.txt

--
Some tips to avoid problems in future:
Always keep in sync with upstream master!
Use branches! If possible separate for code and Czech translations. I suspect You will try to make Cistercian code complete before pushing to main project, but translations can be pushed without hesitation.

**Reply by Augustinus-Altovadensis** [reply]
> ad. 2.
> Introducing section like [Ant LaudesC] in monastic data files seems valid with default value to first line of [Ant Vespera].
> 
OK, will do. The easiest thing to do will probably be to reference this section, and if it doesn't exist, simply take the  first line of [Ant Vespera].

> If it is 95% introducing new directories looks unnecessary.
> Try use existing existing Monastic data:
> 
Sure. I wanted to keep it separate, but from the final state point of view, this is more logical. I saw not even the monastic version has the same division of readings etc.

> * use magic comments like "(rubrica Cistercian)" or "(rubrica Cistercian omittitur)"
> * or create files like ex. SanctiM/08-20c.txt
> 
Yes, there will surely be a separate SanctiM/08-20c.txt :-) And also other order founders and saints.

> --
> Some tips to avoid problems in future:
> Always keep in sync with upstream master!
> Use branches! If possible separate for code and Czech translations. I suspect You will try to make Cistercian code complete before pushing to main project, but translations can be pushed without hesitation.

Yes, that's the plan. First make it work and then push to the main project.
That's a good idea, using one branch for translations and another for the code! 

And to the translations: I'm trying to get a permission for using a more recent biblical text than some Public Domain. Then I will upload the biblical texts. 

BTW, is there any convention for naming the languages? I saw Latin is in English, meanwhile the rest is in the original languages without diacritics. In Czech, the word would be "Česky", which currently doesn't work well with the code. But the word without diacritics looks strange...
Or is there any way to use a display name different from the internal one for the system?

**Comment by Augustinus-Altovadensis** [comment]
If someone is interested or curious, the code lives in my repository (accessible to all, I hope), and a quasi-live version can be found here:

https://officium.georgius.net

It's far from perfect, but most of it works. Now it's curious, why the Hymn for Terce in Pentecost (Octave) doesn't show at all...
